python/lsst/meas/astrom/fitSipDistortion.py

```diff
@@ -0,0 +1,420 @@
+
+        This method is called by fitWcs when display debugging is enabled.  It
+        always drops into pdb before returning to allow interactive inspection,
+        and hence it should never be called in non-interactive contexts.
+
+        Parameters
+        ----------
+
+        revFitter : :cpp:class:`lsst::meas::astrom::ScaledPolynomialTransformFitter`
+            Fitter object initialized with `fromMatches` for fitting a "reverse"
+            distortion: the mapping from intermediate world coordinates to
+            pixels.
+        exposure : :cpp:class:`lsst::afw::image::Exposure`
+            An Exposure or other displayable image on which matches can be
+            overplotted.
+        bbox : :cpp:class:`lsst::afw::geom::Box2I`
+            Bounding box of the region on which matches should be plotted.
+        """
+
+        data = revFitter.getData()
+        disp = lsst.afw.display.getDisplay()
+        if exposure is not None:
+            disp.mtv(exposure)
+        elif bbox is not None:
+            disp.mtv(exposure=lsst.afw.image.ExposureF(bbox))
+        else:
+            raise TypeError("At least one of 'exposure' and 'bbox' must be provided.")
+        data = revFitter.getData()
+        srcKey = lsst.afw.table.Point2DKey(data.schema["src"])
+        srcErrKey = lsst.afw.table.CovarianceMatrix2fKey(data.schema["src"], ["x", "y"])
+        refKey = lsst.afw.table.Point2DKey(data.schema["initial"])
+        modelKey = lsst.afw.table.Point2DKey(data.schema["model"])
+        validKey = data.schema.find("valid").key
+        with disp.Buffering():
+            for record in data:
+                colors = ((lsst.afw.display.RED, lsst.afw.display.GREEN)
+                            if record.get(validKey) else
+                            (lsst.afw.display.MAGENTA, lsst.afw.display.CYAN))
+                rx, ry = record.get(refKey)
+                disp.dot("x", rx, ry, size=10, ctype=colors[0])
+                mx, my = record.get(modelKey)
+                disp.dot("o", mx, my, size=10, ctype=colors[0])
+                disp.line([(rx, ry), (mx, my)], ctype=colors[0])
+                sx, sy = record.get(srcKey)
+                sErr = record.get(srcErrKey)
+                sEllipse = lsst.afw.geom.ellipses.Quadrupole(sErr[0, 0], sErr[1, 1], sErr[0, 1])
+                disp.dot(sEllipse, sx, sy, ctype=colors[1])
+        print("Dropping into debugger to allow inspection of display. Type 'continue' when done.")
+        import pdb
+        pdb.set_trace()
+
+    def makeInitialWcs(self, matches, wcs):
+        """Generate a guess Wcs from the astrometric matches
+
+        We create a Wcs anchored at the center of the matches, with the scale
+        of the input Wcs.  This is necessary because matching returns only
+        matches with no estimated Wcs, and the input Wcs is a wild guess.
+        We're using the best of each: positions from the matches, and scale
+        from the input Wcs.
+
```

Comment:
I actually just stole this method (including that doc text) from the old WCS fitter task.  If it's no longer accurate, I'll change it.